### PR TITLE
SYS-1671: Security fixes for Digital Signage Hours

### PR DESCRIPTION
**Pull request by ztucker4**
Implements [SYS-1671](https://uclalibrary.atlassian.net/browse/SYS-1671)
Includes these updates, with no functionality changes:

* Base image updated to use Debian 12 and Python 3.12
* Django updated to 4.2.16
  * gunicorn (23.0.0) and requests (2.32.3) also updated to address Dependabot alerts
* Removed version declaration from docker-compose files
* `.docker-compose_django.env` - used single quotes around DJANGO_SECRET_KEY, as was needed for terra
* Updated entrypoint script to actually load locations fixture in dev environment
* `build_and_test` and `build_docker_hub` are updated to use the latest versions of GitHub Actions components, and to add the `scan` tag to the Docker Hub build

I didn't see any errors, warnings, or other concerning messages in the build logs. All 22 tests pass (`docker compose exec django python manage.py test`). Log in with admin/admin works for both the [Application](http://127.0.0.1:8000/) and [Admin](http://127.0.0.1:8000/admin) interfaces, and you should be able to generate working hours display URLs and see correct data from LibCal. I also briefly tested the `gunicorn` upgrade by setting `DJANGO_RUN_ENV=prod` in `.docker-compose_django.env`, restarting, and confirming that the application and logs still look good.

[SYS-1671]: https://uclalibrary.atlassian.net/browse/SYS-1671?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -19,4 +18,5 @@ services:
     volumes:
       - pg_data:/var/lib/postgresql/data/
 volumes:
-  pg_data:
\ No newline at end of file
+  pg_data:
+
```
Comment (akohler) [inline docker-compose.yml:22]: Final line has a single space on it, which should be removed (just keeping the empty line itself).

File: docker_scripts/entrypoint.sh [hunk 2]
```diff
@@ -23,7 +23,7 @@ if [ "$DJANGO_RUN_ENV" = "dev" ]; then
   # Load fixtures, only in dev environment.
   # TODO: Configure or delete
```
Comment (akohler) [inline docker_scripts/entrypoint.sh:24]: This `TODO` line should come out now that it's TODONE.

**Review by akohler — CHANGES_REQUESTED** [review]
@ztucker4 Please update the rest of the Actions versions for `build_docker_hub.yml`:
```
        uses: actions/checkout@v4
        uses: docker/setup-buildx-action@v3
        uses: docker/login-action@v3
```
EDIT: I see `actions/checkout` is up to version 4 now, I missed that with TERRA - please go to v4.

A couple of other very minor tweaks as well, see comments for those.

Everything else looks great.  Application builds cleanly (and quickly now, since it reuses several existing layers from other 3.12 applications).  It loads the sample data, works as expected, tests all pass.

Thanks --Andy

**Comment by ztucker4** [comment]
Changes made, ready for a second look. Thanks!

**Review by akohler — APPROVED** [review]

**Comment by akohler** [comment]
@ztucker4 Thanks for the fixes - merged.

**Pull request merged**